lib/ruby_terraform/commands/base.rb

```diff
@@ -5,16 +5,17 @@ module Commands
     class Base
       attr_reader :binary
 
-      def initialize(binary: nil)
+      def initialize(binary: nil, **log_opts)
         @binary = binary || RubyTerraform.configuration.binary
+        @logger = log_opts[:logger] || RubyTerraform.logger(log_opts[:log_path])
       end
```

Comment:
If you follow the approach outlined above, the pattern for looking up the logger here can be the same as used for the binary.